### PR TITLE
[Merged by Bors] - Documentation

### LINKED ISSUE #40: Documentation
As a user of the spark-k8s operator I want to have documentation/examples for the following:

- kind cluster set-up for local testing (service account, roles, PV, where to find docker images and how to load them to a local cluster)
- explanation of custom resource fields (and how they map to spark-submit arguments)
- a brief description of the examples
- updated changelog
- what has to be provided in the docker images if the user provides their own
- next steps, TODOs etc.

### PR DESCRIPTION
**Pull request by adwk67**
## Description

Documentation of work-to-date:

- kind cluster set-up for local testing (service account, roles, PV, where to find docker images and how to load them to a local cluster)
- explanation of custom resource fields (and how they map to spark-submit arguments)
- a brief description of the examples
- updated changelog
- what has to be provided in the docker images if the user provides their own
- next steps, TODOs etc.

Fixes #40.

## Review Checklist
- [x] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: docs/modules/ROOT/pages/external_storage.adoc [hunk 1]
```diff
@@ -0,0 +1,66 @@
+=== Overview
+
+The spark-on-kubernetes operator builds and executes a `spark-submit` job. This is run as a job against a spark-compatible image specified in the `SparkApplication` custom resource .yaml (by "spark-compatible" we mean that the image must contain or have classpath-access to whatever libraries, configuration files and environments - e.g. python - that are needed for the job). Images are available from the Stackable repository but can also be provided by the user.
+
+This job calls the standard spark `entrypoint.sh` script that 
+
+- does some processing of the `spark-submit` command (such as writing command line arguments to a `spark.properties` that is local to the driver/executors, and switching from cluster to client mode)
+- creates a driver image
```
Comment (razvan) [inline docs/modules/ROOT/pages/external_storage.adoc:8]:
```suggestion
- creates a driver pod
```

File: docs/modules/ROOT/pages/external_storage.adoc [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+- which in turn creates, executes and cleans up executor images
```
Comment (razvan) [inline docs/modules/ROOT/pages/external_storage.adoc:9]:
```suggestion
- which in turn creates, executes and cleans up executor pods
```

File: docs/modules/ROOT/pages/installation.adoc [hunk 3]
```diff
@@ -57,11 +57,27 @@ docker run \
 
 == Building the operator from source
 
-This operator is written in Rust and is developed against the latest stable Rust release (1.58 at
+This operator is written in Rust and is developed against the latest stable Rust release (1.59 at
```
Comment (razvan) [inline docs/modules/ROOT/pages/installation.adoc:60]: I would get rid of the Rust version. This is a moving target that's not even relevant for users of the operator.

**Review by razvan — APPROVED** [review]

**Review by razvan — APPROVED** [review]
lgtm

**Comment by adwk67** [comment]
bors merge

**Pull request closed without merging**